### ISSUE TITLE
Add support for additional TESS HLSP light curves: DIAMANTE, PATHOS, CDIPS, TASOC

### ISSUE THREAD

**Issue by barentsen** [KeplerGO/lightkurve#918]
This is a follow-up of #912.

Several excellent sets of TESS community light curves are now available at MAST as High Level Science Products.  We already support searching and opening SPOC and QLP products via Lightkurve's `search_lightcurve` function, but it would be very valuable to also support the following recent HLSPs which provide light curve files:
* DIAMANTE
* PATHOS
* CDIPS
* TASOC

The following steps are needed to add support for any of these
1. Add a custom reader function for each product type as `lightkurve/io/*NAME*.py`.
2. Register the custom reader in `lightkurve/io/__init__.py`.
3. Update `lightkurve/io/detect.py` and `lightkurve/io/read.py` to detect the products correctly.
4. Add unit tests.

Help is welcome in the form of PRs.  One for each product, please!

**Comment by stephtdouglas** [comment]
I'm starting to work with the CDIPS and PATHOS light curves, so I can try to add the code for both of those, if nobody else is working on them yet

**Comment by barentsen** [comment]
@stephtdouglas Nobody has started working on this yet, and it would be really cool to add this to Lightkurve.  I would be delighted to review and merge a PR if you were interested in preparing one! 👍 

The existing readers located under `lightkurve/io/xxxxx.py` should give you a pretty good example of what the readers might looks like (assuming they are FITS files as well?)  Let me know if you have questions!

**Comment by stephtdouglas** [comment]
@barentsen I have the PATHOS version done, with unit tests written (https://github.com/stephtdouglas/lightkurve/tree/pathos_reader), but I'm having trouble running pytest on my local machine for some reason. If I open a pull request, will that trigger continuous integration testing?

**Comment by barentsen** [comment]
@stephtdouglas  Yep!  When you open a PR, the tests will run and you will see the results online.  You're welcome to use that mechanism before completing work on a branch.  In that case, people often put "[WIP]" in the PR title, i.e. "Work In Progress".

Alternatively, I'm happy to help debug pytest if you copy/paste the error.

**Comment by stephtdouglas** [comment]
@barentsen I'm working on the CDIPS reader now, and have run into another problem. As best I can tell, `io.generic.read_generic_lightcurve()` does not accept an error_column, but instead looks for a column with the format `f"{flux_column}_err"`. The CDIPS light curves include errors for the raw light curves, but they don't have the expected column format (e.g., 'IRM1' is the raw magnitude, but 'IRE1' is the associated error column). Is it better to just add the errors after the fact, as part of the cdips reader, for now? Or should I add an error_column argument to the generic reader?

They also seem to have a non-standard quality column, that seems to be something about their analysis process rather than correlating with the usual TESS flags. My intent is just to ignore that column for now, unless you think I should do that differently.

**Comment by barentsen** [comment]
@stephtdouglas Thank you for looking into this!!

There is no doubt that we should add `time_column` and `error_column` arguments to the `read_generic_lightcurve` function. 

**However**: if this change has complicated repercussions on the existing readers, then it would 100% be fine to "work around" the lack of these arguments in the CDIPS PR for now, and open a separate issue to deal with the lack of the arguments in the future.  I say this because adding CDIPS support is definitely a higher priority than improving the generic reader behind the scenes.

> They also seem to have a non-standard quality column
> My intent is just to ignore that column for now, unless you think I should do that differently.

It really comes down to what cadences the authors suggest users should or shouldn't use, which will likely require browsing their paper (which I haven't done yet).  It would be acceptable to ignore the quality mask for now unless it is necessary to make the light curve useful.  For example, if certain bad cadences have flux set to "NaN" or "-999", then we would *have* to mask them out or the light curves of would be a pain to use.  This might not be the case for CDIPS however, I'm merely speculating!

If you have time to open a few CDIPS files and glance at the paper, then I'm happy to follow your guidance on which columns to use and/or which quality mask to apply by default, if any!

Thank you!!

**Comment by stephtdouglas** [comment]
@barentsen OK, sounds good! I'll take a more careful look at the CDIPS light curves and the current code, and decide what to do from there on both problems. My semester starts Monday, so I may be a bit slower getting this update together, but I'll be working on it :)

**Comment by barentsen** [comment]
@stephtdouglas Good news: I went ahead and added `time_column` and `flux_err_column` arguments to `read_generic_lightcurve()`. (Make sure to branch off the latest version of `master` if you decide to work on this!)

**Comment by kecnry** [comment]
Can you please add [POLAR](https://archive.stsci.edu/prepds/polar/) to the list (or would you prefer I open a separate issue)?
